openedx/features/course_experience/course_tools.py

```diff
@@ -0,0 +1,52 @@
+"""
+Support for course tool plugins.
+"""
+from openedx.core.lib.api.plugins import PluginManager
+
+# Stevedore extension point namespace
+COURSE_TOOLS_NAMESPACE = 'openedx.course_tool'
+
+
+class CourseTool(object):
+    """
+    This is an optional base class for Course Tool plugins.
+
+    Plugin implementations can choose to subclass CourseTool to get
+    useful default behavior, but it is not a requirement. A course tool
+    plugin should define the following class methods:
+
+     * is_enabled(cls, course_key):
```

Comment:
I would replace this section and simply state that all the methods are required.  Then, add a properly commented method with `raise NotImplementedError()` for methods without a default implementation.

I'd also add an example return value for the icon_classes if you think that makes sense.